documentation/release_notes.markdown

```diff
@@ -14,11 +14,46 @@ canonical: "/puppetdb/latest/release_notes.html"
 [metrics]: ./api/metrics/v1/changes-from-puppetdb-v3.html
 [pqltutorial]: ./api/query/tutorial-pql.html
 
+4.1.2
+-----
+
+PuppetDB 4.1.2 is a backward-compatible bugfix release that speeds up node
+queries, fixes a bug that slowed garbage collection, and improves error logging.
+
+### Bug Fixes and Maintenance
+
+* Queries to the /nodes endpoint should be faster, due to the removal of the
+  index on certnames(latest_report_id).
+  ([PDB-2789](https://tickets.puppetlabs.com/browse/PDB-2789))
+
+* Reverted the drop of constraints while sweeping reports and nodes
+  ([PDB-2477](https://tickets.puppetlabs.com/browse/PDB-2477)), which had slowed
+  down garbage collection.
+
+* Command failure messages now log the certname of the node from which the
+  command originated in order to track problematic commands to individual nodes.
+  ([PDB-2580](https://tickets.puppetlabs.com/browse/PDB-2580))
+
+* At-at exceptions are now logged as errors.
```

Comment:
"Some previously masked garbage collection will now appear in the logs as errors." Maybe?